roles/servicetelemetry/templates/manifest_grafana_v5.j2

```diff
@@ -0,0 +1,95 @@
+apiVersion: grafana.integreatly.org/v1beta1
+kind: Grafana
+metadata:
+  name: {{ ansible_operator_meta.name }}-grafana
+  namespace: {{ ansible_operator_meta.namespace }}
+  labels:
+    dashboards: "stf"
+spec:
+  serviceAccount:
+    metadata:
+      annotations:
+        serviceaccounts.openshift.io/oauth-redirectreference.primary: '{{ grafana_oauth_redir_ref | to_json }}'
+{% if servicetelemetry_vars.graphing.grafana.ingress_enabled is defined and servicetelemetry_vars.graphing.grafana.ingress_enabled %}
+  route:
+    spec:
+      port:
+        targetPort: web
+      tls:
+        termination: reencrypt
+      to:
+        kind: Service
+        name: {{ ansible_operator_meta.name }}-grafana-service
+        weight: 100
+      wildcardPolicy: None
+{% endif %}
+  client:
+    preferIngress: false
+  config:
+    auth:
+      disable_signout_menu: "{{ servicetelemetry_vars.graphing.grafana.disable_signout_menu }}"
+      disable_login_form: "True"
+    auth.anonymous:
+      enabled: "True"
+    auth.proxy:
+      enabled: "True"
+      enable_login_token: "True"
+      header_property: "username"
+      header_name: "X-Forwarded-User"
+    log:
+      level: warn
+      mode: "console"
+  deployment:
```

Comment:
I leaned heavily on this documentation to get all of this working as intended in Grafana Operator v5: https://grafana.github.io/grafana-operator/docs/examples/oauth_proxy/readme/

There are some really good examples there about how to deploy the various components.